src/main/java/net/wasdev/wlp/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -0,0 +1,683 @@
+        String installJarFileSubpath = installJarFile.getParentFile().getName() + File.separator + installJarFile.getName();
+        Map<String, Object> mapBasedInstallKernel = AccessController.doPrivileged(new PrivilegedExceptionAction<Map<String, Object>>() {
+            @SuppressWarnings({ "unchecked", "resource" })
+            @Override
+            public Map<String, Object> run() throws Exception {
+                ClassLoader loader = new URLClassLoader(new URL[] { installJarFile.toURI().toURL() }, null);
+                Class<Map<String, Object>> clazz;
+                clazz = (Class<Map<String, Object>>) loader.loadClass("com.ibm.ws.install.map.InstallMap");
+                return clazz.newInstance();
+            }
+        });
+        if (mapBasedInstallKernel == null){
+            throw new PluginExecutionException("Cannot run install jar file " + installJarFile);
+        }
+
+        // Init
+        mapBasedInstallKernel.put("runtime.install.dir", installDirectory);
+        mapBasedInstallKernel.put("install.map.jar", installJarFileSubpath);
+        debug("install.map.jar: " + installJarFileSubpath);
+        debug("install.kernel.init.code: " + mapBasedInstallKernel.get("install.kernel.init.code"));
+        debug("install.kernel.init.error.message: " + mapBasedInstallKernel.get("install.kernel.init.error.message"));
+        File usrDir = new File(installDirectory, "usr/tmp");
+        mapBasedInstallKernel.put("target.user.directory", usrDir);
+        return mapBasedInstallKernel;
+    }
+    
+    /**
+     * Find latest install map jar from specified directory
+     * 
+     * @return the install map jar file
+     */
+    public static File getMapBasedInstallKernelJar(File dir) {
+
+        File[] installMapJars = dir.listFiles(new FilenameFilter() {
+            @Override
+            public boolean accept(File dir, String name) {
+                return name.startsWith(INSTALL_MAP_PREFIX) && name.endsWith(INSTALL_MAP_SUFFIX);
+            }
+        });
+
+        File result = null;
+        if (installMapJars != null) {
+            for (File jar : installMapJars) {
+                if (isReplacementJar(result, jar)) {
+                    result = jar;
+                }
+            }
+        }
+
+        return result;
+    }
+    
+    /**
+     * Returns whether file2 can replace file1 as the install map jar.
+     */
+    private static boolean isReplacementJar(File file1, File file2) {
+        if (file1 == null) {
+            return true;
+        } else if (file2 == null) {
+            return false;
+        } else {
+            String version1 = extractVersion(file1.getName());
+            String version2 = extractVersion(file2.getName());
+            return compare(version1, version2) < 0;
+        }
+    }
+    
+    private static String extractVersion(String fileName) {
+        int startIndex = INSTALL_MAP_PREFIX.length() + 1;
```

Comment:
Why do you add 1 here? If this is to account for the underscore shouldn't that be part of the prefix?

Reply:
Yes.  Added a comment to clarify.